### ISSUE TITLE
Error when modifying service tier config file

### ISSUE THREAD

**Issue by ArkhaLime** [microsoft/navcontainerhelper#2452]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
I get an error with the service tier config file that doesn't exists with BC 20.0.37253.40399.

**Scripts used to create container and cause the issue**
```
$containerName = 'bc20'
$password = '<Password>'
$securePassword = ConvertTo-SecureString -String $password -AsPlainText -Force
$credential = New-Object pscredential 'admin', $securePassword
$auth = 'UserPassword'
$artifactUrl = Get-BcArtifactUrl -type 'Sandbox' -country 'fr' -select 'Latest'
$licenseFile = 'C:\Users\MF\Desktop\BC-20_Exp_2022-08-08.flf'
New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -credential $credential `
    -auth $auth `
    -artifactUrl $artifactUrl `
    -licenseFile $licenseFile `
    -dns '8.8.8.8' `
    -updateHosts `
    -multitenant:$true `
    -myscripts @("https://raw.githubusercontent.com/microsoft/nav-docker/master/override/issue2434/HelperFunctions.ps1") `
    -alwaysPull
```

**Full output of scripts**
```
BcContainerHelper is version 3.0.7
BcContainerHelper is running as administrator
Hyper-V is Enabled
Host is Microsoft Windows 10 Professionnel - 21H2
Docker Client Version is 20.10.14
Docker Server Version is 20.10.14
Removing container bc20
Removing bc20 from container hosts file
Removing bc20-* from container hosts file
Removing C:\ProgramData\BcContainerHelper\Extensions\bc20
Fetching all docker images
Fetching all docker volumes
INFO: Windows 10 21H1/21H2 images are not yet available, using 2004 as these are found to work better than 20H2 on 21H1/21H2
Pulling image mcr.microsoft.com/businesscentral:10.0.19041.1415
10.0.19041.1415: Pulling from businesscentral
Digest: sha256:5081759663a6e853232a99c2f7289d19e384ac148abe637e147d58739fc281e1
Status: Image is up to date for mcr.microsoft.com/businesscentral:10.0.19041.1415
mcr.microsoft.com/businesscentral:10.0.19041.1415
Using image mcr.microsoft.com/businesscentral:10.0.19041.1415
Creating Container bc20
Style: sandbox
Multitenant: Yes
Version: 20.0.37253.40399
Platform: 20.0.37114.40359
Generic Tag: 1.0.2.1
Container OS Version: 10.0.19041.1415 (2004)
Host OS Version: 10.0.19044.1645 (21H2)
WARNING: Host OS is 21H1 and Container OS is 2004, defaulting to process isolation. If you experience problems, add -isolation hyperv.
Using process isolation
Using locale fr-FR
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Downloading C:\ProgramData\BcContainerHelper\Extensions\bc20\my\HelperFunctions.ps1
Using license file C:\Users\MF\Desktop\5888919_LIC-DEV-MERCKEL-FR-ISV-BC-20_Exp_2022-08-08.flf
Files in C:\ProgramData\BcContainerHelper\Extensions\bc20\my:
- AdditionalOutput.ps1
- HelperFunctions.ps1
- license.flf
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container bc20 from image mcr.microsoft.com/businesscentral:10.0.19041.1415
2881d381e4fa25bbb88d23273907e21473a961c6f208685cf28794a6a5b83178
Waiting for container bc20 to be ready
Using artifactUrl https://bcartifacts.azureedge.net/sandbox/20.0.37253.40399/fr
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\20.0.37253.40399\platform\ServiceTier\Program Files
c:\dl\sandbox\20.0.37253.40399\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\20.0.37253.40399\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\20.0.37253.40399\platform\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\sandbox\20.0.37253.40399\fr\BusinessCentral-FR.bak
Restoring CRONUS Demo Database
Initializing...
Setting host.docker.internal to 172.28.165.55 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 172.28.165.55 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.29.16.1 in container hosts file
Starting Container
Hostname is bc20
PublicDnsName is bc20
Using NavUserPassword Authentication
Starting Local SQL Server
WARNING: Waiting for service 'SQL Server (SQLEXPRESS) (MSSQL$SQLEXPRESS)' to
start...
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 6508B0A3C220C1588CB454C7BEFA4D3C4C2FC7C2
DNS identity bc20
Modifying Service Tier Config File with Instance Specific Settings
Service Tier doesn't exist / is not installed
at <ScriptBlock>, C:\Run\navstart.ps1: line 153
at <ScriptBlock>, C:\Run\start.ps1: line 363
at <ScriptBlock>, <No file>: line 1Error
Using artifactUrl https://bcartifacts.azureedge.net/sandbox/20.0.37253.40399/fr
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\20.0.37253.40399\platform\ServiceTier\Program Files
c:\dl\sandbox\20.0.37253.40399\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\20.0.37253.40399\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\20.0.37253.40399\platform\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\sandbox\20.0.37253.40399\fr\BusinessCentral-FR.bak
Restoring CRONUS Demo Database
Initializing...
Setting host.docker.internal to 172.28.165.55 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 172.28.165.55 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.29.16.1 in container hosts file
Starting Container
Hostname is bc20
PublicDnsName is bc20
Using NavUserPassword Authentication
Starting Local SQL Server
WARNING: Waiting for service 'SQL Server (SQLEXPRESS) (MSSQL$SQLEXPRESS)' to
start...
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 6508B0A3C220C1588CB454C7BEFA4D3C4C2FC7C2
DNS identity bc20
Modifying Service Tier Config File with Instance Specific Settings
Service Tier doesn't exist / is not installed
at <ScriptBlock>, C:\Run\navstart.ps1: line 153
at <ScriptBlock>, C:\Run\start.ps1: line 363
at <ScriptBlock>, <No file>: line 1
New-BcContainer Telemetry Correlation Id: 3a275598-3cf3-4be4-b0c9-5725931078c7
Initialization of container bc20 failed
Au caractère C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.7\ContainerHandling\Wait-NavContainerReady.ps1:42 : 17
+ ...             throw "Initialization of container $containerName failed" ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Initialization of container bc20 failed:String) [], RuntimeException
    + FullyQualifiedErrorId : Initialization of container bc20 failed

...
```

**Screenshots**

**Additional context**
The container is created in a clean VM (Win 10 Pro) on my laptop (also Win 10 Pro). The docker on my laptop was several times reset but keep making errors (volumes errors or docker not starting).

In the script, if the parameter "myscripts" is deleted, I get an error like in issue #2439

**Comment by freddydk** [comment]
This looks like yet another example of problems with Windows Desktop and Containers.
https://freddysblog.com/2022/05/10/resilience/

If you try to add this to your new-bccontainer:

-useGenericImage "$(Get-BestGenericImageName)-dev"

and let me know whether it solves the issue?

**Comment by ArkhaLime** [comment]
Hi Freddy,

I could test with the new generic image in the VM but I got an error about disk space a little after the service tier config modification.
But I tested successfully the command on my local Docker 😃 

What is the difference between the production image and the dev image ?

Thank you for the quick response and error resolution!

**Comment by freddydk** [comment]
-dev images are prerelease and will be shipped as production later this week

**Comment by freddydk** [comment]
Generic image 1.0.2.5 has now shipped together with BcContainerHelper 3.0.8
